arches/app/datatypes/core/geojson_feature_collection.py

```diff
@@ -0,0 +1,430 @@
+                         Adjust your coordinates or your settings.DATA_EXTENT_VALIDATION property."
+                    )
+                    title = _("Geometry Out Of Bounds")
+                    errors.append(
+                        {
+                            "type": "ERROR",
+                            "message": "datatype: {0} value: {1} {2} - {3}. {4}".format(
+                                self.datatype_model.datatype,
+                                value,
+                                source,
+                                message,
+                                "This data was not imported.",
+                            ),
+                            "title": title,
+                        }
+                    )
+            except Exception:
+                message = _("Not a properly formatted geometry")
+                title = _("Invalid Geometry Format")
+                errors.append(
+                    {
+                        "type": "ERROR",
+                        "message": "datatype: {0} value: {1} {2} - {3}. {4}.".format(
+                            self.datatype_model.datatype,
+                            value,
+                            source,
+                            message,
+                            "This data was not imported.",
+                        ),
+                        "title": title,
+                    }
+                )
+
+        if value is not None:
+            for feature in value["features"]:
+                try:
+                    geom = GEOSGeometry(JSONSerializer().serialize(feature["geometry"]))
+                    if geom.valid:
+                        validate_geom_bbox(geom)
+                    else:
+                        raise Exception
+                except Exception:
+                    message = _("Unable to serialize some geometry features.")
+                    title = _("Unable to Serialize Geometry")
+                    error_message = self.create_error_message(
+                        value, source, row_number, message, title
+                    )
+                    errors.append(error_message)
+        return errors
+
+    def to_json(self, tile, node):
+        data = self.get_tile_data(tile)
+        if data:
+            return self.compile_json(tile, node, geojson=data.get(str(node.nodeid)))
+
+    def clean(self, tile, nodeid):
+        if tile.data[nodeid] is not None and "features" in tile.data[nodeid]:
+            if len(tile.data[nodeid]["features"]) == 0:
+                tile.data[nodeid] = None
+
+    def check_geojson_value(self, value):
+        if type(value) is str:
+            geojson = json.loads(value)
+        else:
+            geojson = value
+
+        features = []
+        if geojson["type"] == "FeatureCollection":
+            for feature in geojson["features"]:
+                if feature["geometry"]["type"].find("Multi") != -1:
```

Comment:
With Python strings you can check for string patterns directly: ` if "Multi" in feature["geometry"]["type"]:`

Reply:
old habits...